brambling/models.py

```diff
@@ -1342,13 +1342,16 @@ def from_dwolla_refund(cls, refund, related_transaction, **kwargs):
     def is_unconfirmed_check(self):
         return self.method == Transaction.CHECK and not self.is_confirmed
 
-    def can_refund(self):
+    def get_returnable_items(self):
+        return self.bought_items.filter(status=BoughtItem.BOUGHT)
+
+    def get_refundable_amount(self):
         refunded = self.related_transaction_set.filter(
             transaction_type=Transaction.REFUND
         ).aggregate(refunded=Sum('amount'))['refunded']
         # None means there are no refunds, which is relevant
         # for 0-amount transactions.
-        return refunded is None or self.amount + refunded > 0
+        return self.amount if refunded is None else self.amount + refunded
 
     def refund(self, amount=None, bought_items=None, issuer=None, dwolla_pin=None):
```

Comment:
The beginning of refund should probably be reworked in a couple ways:

- use self.get_refundable_amount()
- Don't return None if there isn't a refundable amount (is there anything out there relying on this behavior?)
- use self.get_refundable_items() for default bought_items.
- default amount to refundable amount if not provided.